src/Native/Unix/System.Native/pal_time.c

```diff
@@ -152,3 +153,75 @@ int32_t SystemNative_GetTimebaseInfo(uint32_t* numer, uint32_t* denom)
     }
     return 1;
 }
+
+static long numProcessors = 0;
+int32_t SystemNative_GetCpuUtilization(ProcessCpuInformation* previousCpuInfo)
+{
+    if (numProcessors <= 0)
+    {
+        numProcessors = sysconf(_SC_NPROCESSORS_CONF);
```

Comment:
In coreclr we use _SC_NPROCESSORS_CONF only on ARM and _SC_NPROCESSORS_ONLN everywhere else:
https://github.com/dotnet/coreclr/blob/030a3ea9b8dbeae89c90d34441d4d9a1cf4a7de6/src/pal/src/misc/sysinfo.cpp#L107-L113
Should we be doing the same thing here?
https://github.com/dotnet/coreclr/pull/18289

Reply:
Ah, I may have answered my own question: getrusage doesn't let you get the data for an arbitrary process by ID.

Reply:
That's tricky. This added function doesn't handle dynamically added/removed processors. It is common in big/little ARM machines to disable cores which I assume is the reason for the special case in CoreCLR. On some server machines it is possible to hot swap CPUs and in that case this function will likely return incorrect (low) values, but still within the valid range. I will need to read more of the code to provide a satisfactory answer...

Reply:
I guess I will go with your suggestion to match CoreCLR. This new function is nearly verbatim copy of https://github.com/dotnet/coreclr/blob/a082f4702540309e832c2ebdf86b15dc798b136b/src/pal/src/thread/process.cpp#L2461-L2576, which ends up calling `PAL_GetLogicalCpuCountFromOS`. If  it's good enough for CoreCLR then it should be good enough here.